### DISCUSSION TITLE
USBHIDKeyboard boot protocol

### DISCUSSION THREAD

**Discussion by aberoev** [espressif/arduino-esp32#9281] [General]
[Seeing this commit ](https://github.com/espressif/arduino-esp32/commit/9a9ec09f63adf35b132994cec5fae2e7eacdd1df) I was very hopeful to start making my ESP32S3 dev module work as a keyboard emulator for my hardware sound mixer, but no dice. Everything compiles and works great in Windows 10, but not on the hardware sound mixer which requires a boot protocol keyboard. I have tested the sound mixer with regular keyboards and with a atmega32u4 / [Nicohood HID project Bootkeyboard](https://github.com/NicoHood/HID/blob/master/examples/Keyboard/BootKeyboard/BootKeyboard.ino) contraption which works.

Using 3.0.0Alpha3

Hoping for some hints. Using a Lilygo T-QT with these settings

![image](https://github.com/espressif/arduino-esp32/assets/155575523/fa300012-f6ea-4f87-a632-cfa37271774d)

**Comment by aberoev** [comment]
Here's my sketch

```

#include "USB.h"
#include "USBHIDKeyboard.h"
USBHIDKeyboard Keyboard;


void setup() {
  pinMode(0, INPUT_PULLUP);
  Keyboard.begin();
  USB.begin();
}

void loop() {
  while (digitalRead(0) == HIGH) {
    // do nothing until pin 2 goes low

  }
      Keyboard.press(KEY_UP_ARROW);
      delay(100);
      Keyboard.releaseAll();
 
}

```

**Comment by steffen9000** [comment]
Hi aberoev,

did you ever figure out if there is a way to solve this using the ESP32S3? I am trying to control a KVM switch by emulating keyboard shortcuts to toggle between inputs, but apparently, the switch also requires a boot protocol keyboard.

**Reply by GarrettBlackmon** [reply]
Same man, Did you find a way?

You can do it with a picoW according to this thread.
https://forum.level1techs.com/t/controlling-kvm-via-circuitpython-esp32-s3-hid-emulation/212130/2

If I cant figure it out soon I'll probably go this route.
